Assets/Scripts/Controllers/BuildModeController.cs

```diff
@@ -246,6 +251,56 @@ public bool DoesBuildJobOverlapExistingBuildJob(Tile t, string furnitureType)
         return false;
     }
 
+    public void DeconstructFurnitureAt(Tile t, string nextBuild = null)
+    {
+        Job j;
+
+        if (PrototypeManager.FurnitureJobDeconstruct.HasPrototype(t.Furniture.ObjectType))
+        {
+            // Make a clone of the job prototype
+            j = PrototypeManager.FurnitureJobDeconstruct.GetPrototype(t.Furniture.ObjectType).Clone();
+
+            // Assign the correct tile.
+            j.tile = t;
+            j.OnJobCompleted += (Job jo) =>
+            {
+                t.Furniture.Deconstruct();
```

Comment:
This would lead to `j` not collected by GC you need to detach after you finished.